mysite/unmasque/test/experiments/NestingDetectionTest.py

```diff
@@ -0,0 +1,87 @@
+        self.conn.config.detect_nep = False
+        self.conn.config.detect_or = False
+        self.conn.config.detect_oj = False
+        factory = PipeLineFactory()
+        self.pipeline = factory.create_pipeline(self.conn)
+
+    def check_assert(self, truth):
+        print(truth)
+        if truth is None:
+            self.assertTrue(False)
+        self.assertTrue(truth)
+
+    def test_nested_sum(self):
+        query = "select c_name, c_acctbal from customer " \
+                "where (select sum(o_totalprice) as total_sum " \
+                "from orders where c_custkey = o_custkey " \
+                "and o_orderstatus = 'O') " \
+                "< 120000  and c_mktsegment = 'BUILDING' limit 5;"
+        eq = self.do_test(query)
+        self.assertTrue('SUM(o_totalprice)' in eq)
+
+    def test_nested_sum_from_clause(self):
+        query = "select c_name, c_acctbal from customer,  " \
+                 "(select o_custkey, sum(o_totalprice) as total_sum " \
+                 "from orders where o_orderstatus = 'O' group by o_custkey) as avgTable " \
+                 "where avgTable.o_custkey = c_custkey and " \
+                 "avgTable.total_sum < 12000  and c_mktsegment = 'BUILDING' limit 5;"
+        eq = self.do_test(query)
+        self.assertTrue('SUM(o_totalprice)' in eq)
+        self.assertTrue(self.pipeline.correct)
+
+
+    def test_nested_avg(self):
+        query = "select c_name, c_acctbal from customer " \
+                "where (select avg(o_totalprice) as custom_avg " \
+                "from orders where c_custkey = o_custkey " \
+                "and o_orderstatus = 'O') " \
+                "< 120000  and c_mktsegment = 'BUILDING' limit 5;"
+        eq = self.do_test(query)
+        self.assertTrue('AVG(o_totalprice)' in eq)
+
+    def test_nested_min(self):
+        query = "select c_name, c_acctbal from customer " \
+                "where (select min(o_totalprice) as total_sum " \
+                "from orders where c_custkey = o_custkey " \
+                "and o_orderstatus = 'O') " \
+                "< 120000  and c_mktsegment = 'BUILDING' limit 5;"
+        eq = self.do_test(query)
+        self.assertTrue('MIN(o_totalprice)' in eq)
+
+    def do_test(self, query):
+        eq = self.pipeline.doJob(query)
+        self.pipeline.time_profile.print()
+        self.check_assert(eq)
+        return eq
+
+    def test_nested_max(self):
+        query = "select c_name, c_acctbal from customer " \
+                "where (select max(o_totalprice) as custom_avg " \
+                "from orders where c_custkey = o_custkey " \
+                "and o_orderstatus = 'O') " \
+                "< 120000  and c_mktsegment = 'BUILDING' limit 5;"
+        eq = self.do_test(query)
+        self.assertTrue('MAX(o_totalprice)' in eq)
+
+    def test_nested_sum_2_outer_tables(self):
+        query = "select c_name, c_acctbal, n_name from customer, nation " \
```

Comment:
with cte_lps as (select avg(l_extendedprice) as revenue
	from lineitem, partsupp, part 
	where l_partkey = ps_partkey and l_partkey = p_partkey and 
	(l_quantity >= ps_availqty or l_discount >= p_retailprice))
select l_shipmode, ps_supplycost
	from lineitem, partsupp, part, cte_lps
	where l_partkey = ps_partkey and ps_partkey = p_partkey and
	p_container IN ('MED CASE', 'JUMBO JAR')
	and l_extendedprice >= cte_lps.revenue;

Reply:
with cte_orders as 
	(select o_custkey, 
	sum(o_totalprice) as total_sum 
	from orders 
	where o_orderstatus = 'O' 
	group by o_custkey)
select c_name, c_acctbal from customer, cte_orders
	where cte_orders.o_custkey = c_custkey 
	and cte_orders.total_sum < 12000 
	and c_mktsegment = 'BUILDING' 
	order by c_acctbal limit 10;

with subq as 
	(select l_orderkey, sum(l_extendedprice) as total_sum 
	from lineitem 
	where l_linenumber = 4 
	group by l_orderkey)
select o_clerk, o_totalprice 
	from orders,  subq 
	where subq.l_orderkey = o_orderkey 
	and subq.total_sum < 12000  
	and o_orderpriority = '1-URGENT' 
	order by o_totalprice limit 10;